pkg/controller/argocd/role.go

```diff
@@ -116,3 +117,12 @@ func (r *ReconcileArgoCD) reconcileClusterRole(name string, policyRules []v1.Pol
 	applyReconcilerHook(cr, clusterRole)
 	return clusterRole, r.client.Update(context.TODO(), clusterRole)
 }
+
+func deleteClusterRoles(c client.Client, clusterRoleList *v1.ClusterRoleList) error {
+	for _, clusterRole := range clusterRoleList.Items {
+		if err := c.Delete(context.TODO(), &clusterRole); err != nil {
+			return err
```

Comment:
Might want to return return `fmt.Errorf("failed to delete ClusterRole %q during cleanup: %w", clusterRole.Name, err)` in case the returned error doesn't indicate it.

Reply:
Updated the error message